### PR TITLE
[master] Docs for master cluster

### PR DESCRIPTION
**Pull request by dwoz**
### What issues does this PR fix or reference?
Fixes:

### Previous Behavior
Remove this section if not relevant

### New Behavior
Remove this section if not relevant

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [ ] Docs
- [ ] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [ ] Tests written/updated

### Commits signed with GPG?
Yes/No

Please review [Salt's Contributing Guide](https://docs.saltproject.io/en/master/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

File: doc/ref/configuration/master.rst [hunk 1]
```diff
@@ -193,6 +193,50 @@ following the Filesystem Hierarchy Standard (FHS) might set it to
...
 
+
+.. conf_master:: cluster_id
+
+``cluster_id``
+--------------
+
+.. versionadded:: 3007
+
+When defined the master will operate in cluster mode. The master will send the cluster key and id to minions instead of it's own key and id. The master will also forward it's local event bus to other master defined by ``cluster_peers``
```
Comment (twangboy) [inline doc/ref/configuration/master.rst:204]: 1. When defined`,` the master will...
2. ...other master`s` defined by \`\`cluster_peers\`\`

File: doc/ref/configuration/master.rst [hunk 2]
```diff
@@ -193,6 +193,50 @@ following the Filesystem Hierarchy Standard (FHS) might set it to
...
+    cluster_id: master
+
+.. conf_master:: cluster_peers
+
+``cluster_peers``
+-----------------
+
+.. versionadded:: 3007
+
+When ``cluster_id`` is defined. This setting is a list of other master (hostnames or ips) that will be in the cluster.
```
Comment (twangboy) [inline doc/ref/configuration/master.rst:218]: When \`\`cluster_id\`\` is defined`, this` setting is a list of other master`s` ...`

File: doc/ref/configuration/master.rst [hunk 3]
```diff
@@ -193,6 +193,50 @@ following the Filesystem Hierarchy Standard (FHS) might set it to
...
+       - master3
+
+.. conf_master:: cluster_pki_dir
+
+``cluster_pki_dir``
+-------------------
+
+.. versionadded:: 3007
+
+When ``cluster_id`` is defined. This sets the location of where this cluster will store it's cluster publich and private key as well as any minion keys. This setting will default to the value of ``pki_dir``, but should be changed filesystem location shared between peers in the cluster.
```
Comment (twangboy) [inline doc/ref/configuration/master.rst:233]: 1. When  \`\`cluster_id\`\` is defined `, this` sets the location ...
2. ...where this cluster will store `its` cluster `public` and private key ...
3. ...should be changed `to the` filesystem location...

File: doc/topics/highavailability/index.rst [hunk 4]
```diff
@@ -8,6 +8,21 @@ Salt supports several features for high availability and fault tolerance.
...
 
+
+Master Cluster
+==============
+
+.. versionadded:: 3007
+
+Salt masters can be configured to act as a cluster. All masters in a cluster
+are peers. Job workloads are shared accross the cluster. Master clusters
+provide a way to scale masters horrizontally. They do not require changes to
```
Comment (twangboy) [inline doc/topics/highavailability/index.rst:19]: `horizontally`

File: doc/topics/tutorials/master-cluster.rst [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+must all be on a shared filesystem. Most implimentations will also server the
+masters publish and request server ports via a tcp load balancer. All of the
+masters in a cluster are assumed to be running on a reliable local area
+network.
+
+Each master in a cluster maintains it's own public and private key, and an in
+memory aes key. Each cluster peer also has access to the `cluster_pki_dir`
+where a cluster wide public and private key are stored. In addition the cluster
+wide aes key is generated and stored in the `cluster_pki_dir`. In addition,
+when operation as a cluster, minion keys are stored in the `cluster_pki_dir`
```
Comment (twangboy) [inline doc/topics/tutorials/master-cluster.rst:31]: ... when `operating` as a cluster ...

**Review by twangboy — CHANGES_REQUESTED** [review]

File: doc/ref/configuration/master.rst [hunk 6]
```diff
@@ -193,6 +193,56 @@ following the Filesystem Hierarchy Standard (FHS) might set it to
...
+
+.. conf_master:: cluster_id
+
+``cluster_id``
+--------------
+
+.. versionadded:: 3007
+
+When defined, the master will operate in cluster mode. The master will send the
+cluster key and id to minions instead of it's own key and id. The master will
```
Comment (twangboy) [inline doc/ref/configuration/master.rst:205]: its

File: doc/ref/configuration/master.rst [hunk 7]
```diff
@@ -193,6 +193,56 @@ following the Filesystem Hierarchy Standard (FHS) might set it to
...
+also forward it's local event bus to other masters defined by ``cluster_peers``
```
Comment (twangboy) [inline doc/ref/configuration/master.rst:206]: its

File: doc/topics/highavailability/index.rst [hunk 8]
```diff
@@ -8,6 +8,22 @@ Salt supports several features for high availability and fault tolerance.
...
+Master Cluster
+==============
+
+.. versionadded:: 3007
+
+Salt masters can be configured to act as a cluster. All masters in a cluster
+are peers. Job workloads are shared accross the cluster. Master clusters
+provide a way to scale masters horizontally. They do not require changes to
+the minions' configuration to add more resources. Cluster implementations are
+expected to use a load balancer, shared filesystem, and run on areliable
```
Comment (twangboy) [inline doc/topics/highavailability/index.rst:21]: a reliable

Comment (dmurphy18) [inline doc/topics/highavailability/index.rst:21]: areliable -> a reliable

**Review by twangboy — CHANGES_REQUESTED** [review]

File: doc/topics/tutorials/master-cluster.rst [hunk 9]
```diff
@@ -0,0 +1,98 @@
...
+must all be on a shared filesystem. Most implementations will also serve the
+masters publish and request server ports via a tcp load balancer. All of the
+masters in a cluster are assumed to be running on a reliable local area
+network.
+
+Each master in a cluster maintains its own public and private key, and an in
+memory aes key. Each cluster peer also has access to the `cluster_pki_dir`
+where a cluster wide public and private key are stored. In addition, the cluster
+wide aes key is generated and stored in the `cluster_pki_dir`. Further,
+when operation as a cluster, minion keys are stored in the `cluster_pki_dir`
```
Comment (dmurphy18) [inline doc/topics/tutorials/master-cluster.rst:31]: when operated as a cluster

Or Shane's resolved comment, reads as more correct English

**Review by dmurphy18 — CHANGES_REQUESTED** [review]
spelling, English usage

**Review by twangboy — APPROVED** [review]

**Pull request merged**